### PR TITLE
modifies the URL which are output so that they include the timestamp …

### PR DESCRIPTION
**Pull request by johnspackman**
…of the file; this allows strong (even immutable) caching of source files without stale content

### CODE REVIEW

File: source/class/qx/tool/compiler/targets/meta/AbstractJavascriptMeta.js [hunk 1]
```diff
@@ -1,3 +1,25 @@
+/* ************************************************************************
+ *
+ *    qooxdoo-compiler - node.js based replacement for the Qooxdoo python
+ *    toolchain
+ *
+ *    https://github.com/qooxdoo/qooxdoo-compiler
```
Comment (derrell) [inline source/class/qx/tool/compiler/targets/meta/AbstractJavascriptMeta.js:6]: This isn't right any longer, is it? (Might be a global change that should be made in a separate PR.)

**Review by derrell — APPROVED** [review]

**Review by oetiker — APPROVED** [review]

**Review by level420 — APPROVED** [review]

**Review by hkollmann — APPROVED** [review]

File: source/class/qx/tool/compiler/targets/meta/AbstractJavascriptMeta.js (hunk 1 above)
Comment (johnspackman) [reply source/class/qx/tool/compiler/targets/meta/AbstractJavascriptMeta.js:6]: yes - rather than fix it a bit here and a bit elsewhere, I've done a fix-it-everywhere PR in https://github.com/qooxdoo/qooxdoo/pull/10249

**Comment by johnspackman** [comment]
The last few commits include a fix where the webfont sources can be overridden by the application - eg the application includes the paid-for FontAwesome Pro, where the package/framework only use the open source FontAwesome Free, in which case the font name will be the same, but the font files are different.

**Review by cboulanger — APPROVED** [review]

**Review by hkollmann — CHANGES_REQUESTED** [review]
Checks are not working any longer. Could you repair them?
Thanks!

**Comment by cboulanger** [comment]
If this is a node10 problem, I suggest we remove node 10

**Comment by hkollmann** [comment]
No it's a problem with the new parameter

**Comment by johnspackman** [comment]
lint error, curse of the curly braces.  i'll fix it shortly.

**Comment by hkollmann** [comment]
Hi John,
problem is this test in test-deps.js: 
src.match(/src="appone\/index.js"/)

the newly created index.html file contains 

src="appone/index.js${indexJsTimestamp}"

should i fix the test in your PR?

**Comment by johnspackman** [comment]
@hkollmann np, just done it

**Review by hkollmann — APPROVED** [review]
